### ISSUE TITLE
Unable to load any of the alternatives:("libSDL2-2... in android

### ISSUE THREAD

**Issue by noloop** [lispgames/cl-sdl2#126]
I loaded the libSDL2.so which I compile to armv7a and l loaded with:

`(cffi:load-foreign-library #P"/data/data/ecl.android.skeleton/lib/libSDL2.so") ` => its ok, I checked

but when I load the sdl2 system:

`(asdf:load-system :sdl2)`

I get:

```
Unable to load any of the alternatives:
("libSDL2-2.0.so.0" "libSDL2.so.0.2" "libSDL2")
```

How do I get to `libSD2.so` for cl-sdl2 to recognize?

**Comment by notmgsk** [comment]
@noloop `cffi:load-system` takes a `:search-path` (a list of non-default paths to search for the library).

**Comment by noloop** [comment]
@notmgsk I tried `(cffi:load-system :sdl2 :search-path #P"/data/data/ecl.android.skeleton/lib/")` but doesn't work.

**Comment by noloop** [comment]
I think that the problem is: https://github.com/rpav/cl-autowrap/issues/98

**Comment by noloop** [comment]
I'm almost can make the SDL2 run in the Android. I have add somethings in the `cl-autowrap` to make it possible. But I get the error:

```common-lisp
ERROR MSG: In method definition for PRINT-OBJECT, found an invalid specializer (SDL-POINT T)
```

I don't understand what can be this yet. Some ideia?

**Comment by noloop** [comment]
> I don't understand what can be this yet. Some ideia?

The libSDL2.so was not loading correctly, I get the following error in ECL:

```
An error occurred during initialization:
LOAD-FOREIGN-MODULE: Could not load foreign module "/home/noloop/Programs/ecl-cross-compile-libs/libSDL2.so" (Error: "/home/noloop/Programs/ecl-cross-compile-libs/libSDL2.so: cannot open shared object file: No such file or directory").
```

**Comment by hu-moonstone** [comment]
```
invalid specializer (SDL-POINT T)
```
This issue was addressed recently by ECL. I think the latest development branch has solved it. I have confirmed that it has been solved on x86 GNU/Linux.

https://gitlab.com/embeddable-common-lisp/ecl/issues/553

**Comment by noloop** [comment]
@hu-moonstone  Yes, I can load the cl-sdl2 with the asdf now, however there are some problems to resolver yet, like the `with-init` that doesn't work because of threads questions, the cl-sdl2 in the Android need use one thread created by the JNI or Java code, so it will need make new functions that doesn't create threads.
